Mathlib/LinearAlgebra/Vandermonde.lean

```diff
@@ -106,70 +114,71 @@ theorem det_vandermonde {n : ℕ} (v : Fin n → R) :
-      rw [Finset.sum_range_succ, add_comm, tsub_self, pow_zero, mul_one, Finset.mul_sum]
-      congr 1
-      refine' Finset.sum_congr rfl fun i' hi' => _
-      rw [mul_left_comm (v 0), Nat.succ_sub, pow_succ]
-      exact nat.lt_succ_iff.mp (finset.mem_range.mp hi')
+    _ = (Finset.prod Finset.univ (fun i => v (Fin.succ i) - v 0)) *
+    det fun i j : Fin n => v (Fin.succ i) ^ (j : ℕ) := by
+      refine' (congr_arg _ _)
+      · cases n
+        · rw [det_eq_one_of_card_eq_zero (Fintype.card_fin 0),
+          det_eq_one_of_card_eq_zero (Fintype.card_fin 0)]
+        apply det_eq_of_forall_col_eq_smul_add_pred fun _ => v 0
+        · intro j
+          simp
+        · intro i j
+          simp only [smul_eq_mul, Pi.add_apply, Fin.val_succ, Fin.coe_castSucc, Pi.smul_apply]
+          rw [Finset.sum_range_succ, add_comm, tsub_self, pow_zero, mul_one, Finset.mul_sum]
+          congr 1
+          refine' Finset.sum_congr rfl fun i' hi' => _
+          rw [mul_left_comm (v 0), Nat.succ_sub, pow_succ]
+          exact Nat.lt_succ_iff.mp (Finset.mem_range.mp hi')
+    _ = ∏ i : Fin n.succ, Finset.prod (Ioi i) (fun j => v j - v i) := by
+      simp_rw [Fin.prod_univ_succ, Fin.prod_Ioi_zero, Fin.prod_Ioi_succ]
+      have h := ih (v ∘ Fin.succ)
+      unfold Function.comp at h
+      rw [h]
 #align matrix.det_vandermonde Matrix.det_vandermonde
 
 theorem det_vandermonde_eq_zero_iff [IsDomain R] {n : ℕ} {v : Fin n → R} :
     det (vandermonde v) = 0 ↔ ∃ i j : Fin n, v i = v j ∧ i ≠ j := by
   constructor
   · simp only [det_vandermonde v, Finset.prod_eq_zero_iff, sub_eq_zero, forall_exists_index]
-    exact fun i _ j h₁ h₂ => ⟨j, i, h₂, (mem_Ioi.mp h₁).ne'⟩
+    rintro i ⟨_, j, h₁, h₂⟩
+    exact ⟨j, i, h₂, (mem_Ioi.mp h₁).ne'⟩
   · simp only [Ne.def, forall_exists_index, and_imp]
     refine' fun i j h₁ h₂ => Matrix.det_zero_of_row_eq h₂ (funext fun k => _)
     rw [vandermonde_apply, vandermonde_apply, h₁]
 #align matrix.det_vandermonde_eq_zero_iff Matrix.det_vandermonde_eq_zero_iff
 
 theorem det_vandermonde_ne_zero_iff [IsDomain R] {n : ℕ} {v : Fin n → R} :
     det (vandermonde v) ≠ 0 ↔ Function.Injective v := by
-  simpa only [det_vandermonde_eq_zero_iff, Ne.def, not_exists, not_and, Classical.not_not]
+  unfold Function.Injective
+  simp only [det_vandermonde_eq_zero_iff, Ne.def, not_exists, not_and, Classical.not_not]
 #align matrix.det_vandermonde_ne_zero_iff Matrix.det_vandermonde_ne_zero_iff
 
 theorem eq_zero_of_forall_index_sum_pow_mul_eq_zero {R : Type _} [CommRing R] [IsDomain R] {n : ℕ}
     {f v : Fin n → R} (hf : Function.Injective f)
     (hfv : ∀ j, (∑ i : Fin n, f j ^ (i : ℕ) * v i) = 0) : v = 0 :=
   eq_zero_of_mulVec_eq_zero (det_vandermonde_ne_zero_iff.mpr hf) (funext hfv)
-#align matrix.eq_zero_of_forall_index_sum_pow_mul_eq_zero Matrix.eq_zero_of_forall_index_sum_pow_mul_eq_zero
+#align matrix.eq_zero_of_forall_index_sum_pow_mul_eq_zero
+Matrix.eq_zero_of_forall_index_sum_pow_mul_eq_zero
```

Comment:
I remember there's some script which expects #align statements to be all on one line, and there's an exception in the long line linter for exactly this. Could you put these back on one line?